kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/SchemaUtils.java

```diff
@@ -79,6 +82,118 @@ public static void validateSchema(StructType schema, boolean isColumnMappingEnab
     validateSupportedType(schema);
   }
 
+  /**
+   * Performs the following validations on an updated table schema:
+   *
+   * <p>The following checks are performed:
+   *
+   * <ul>
+   *   <li>No duplicate columns are allowed
+   *   <li>Column names contain only valid characters
+   *   <li>Data types are supported
+   *   <li>No new non-nullable fields are added
+   *   <li>Physical column name consistency is preserved in the new schema
+   *   <li>Partition columns are not modified
+   *   <li>Nested IDs for array/map types are preserved in the new schema
+   *   <li>No type changes
+   * </ul>
+   */
+  public static void validateUpdatedSchema(
+      StructType currentSchema,
+      StructType newSchema,
+      List<String> partitionCols,
+      Metadata metadata) {
+    validateSchema(newSchema, true /*columnMappingEnabled*/);
```

Comment:
when should callers invoke `validateSchema` (which is a public utility within kernel internals) vs `validateUpdatedSchema` ?

(i.e. these names are a bit vague ..)

Reply:
I updated the JavaDoc to say "using the current schema as a base for validation." I'd say that's the distinction. This function is given a prior schema and using that metadata as a source of truth for validation on the changes done to the updated schema. 

`validateSchema` at the moment is validating no duplicate field names, valid characters and supported data types? Maybe we can rename it to `validateColumnNamesAndTypes`?

I'd also probably also make`validateColumnNamesAndTypes` private and make validateUpdatedSchema the entry point. For a new table we can pass null or optional for the previous and then only call validateColumnNamesAndTypes, and the updating schema case we'd go through the code path in this PR as well. I'd say we can tackle it separately.